backend/courses/views.py

```diff
@@ -246,13 +263,22 @@ def get_queryset(self):
                 ),
             )
         )
+        check_offered_in = self.request.query_params.get("check_offered_in")
```

Comment:
Why use check offered in instead of just changing the semester slug param?

Reply:
Changing the semester slug param will change the returned data, whereas check offered in is just a guard that will return 404 if the course wasn't offered in a certain semester, but won't change the returned data